build.ps1

```diff
@@ -109,6 +109,25 @@ $gitCommit = git rev-parse HEAD
 git archive --format=zip -o "$OUTPUT_DIR\\dotnet-packages-sources.zip" $gitCommit
 "dotnet-packages.git $gitCommit" | Out-File -FilePath "$OUTPUT_DIR\dotnet-packages-manifest.txt"
 
+# prepare NuGet Config
+
+$RESTORE_NUGET_CONFIG_FILE="NuGet.Config"
+Copy-Item "$REPO\$RESTORE_NUGET_CONFIG_FILE" -Destination "$SCRATCH_DIR\$RESTORE_NUGET_CONFIG_FILE"
```

Comment:
here you've updated the contents of `Nuget.Config` in the scratch, but `RESTORE_NUGET_CONFIG_FILE` is still pointing to the old one

I think to keep things clean you should move the prepping of Nuget.Config before the initial assignment of RESTORE_SWTICHES, and make sure that the path we pass to `-p:RestoreConfigFile` is the one in the scratch directory.

Reply:
happy to have a chat if you need clarifications - DM anytime :D